node/tcp/src/tcp.rs

```diff
@@ -64,7 +66,7 @@ pub struct InnerTcp {
     /// The node's configuration.
     config: Config,
     /// The node's listening address.
-    listening_addr: Option<SocketAddr>,
+    listening_addr: OnceCell<SocketAddr>,
```

Comment:
A question more for my curiosity:

`OnceCell` will work well for our use case here, but I'm curious if `OnceBox` would be a better choice (and whether it makes a difference or not). Again, more for my learning than an actual change request here.

Reply:
It's a bit more strict, and since I haven't digested all the new logic yet, I wanted to make sure that the cell doesn't get initialized twice with these changes.

Now that we've confirmed that this doesn't happen, it could indeed be `OnceBox` instead (though it's unlikely that we'll see any difference).